### DISCUSSION TITLE
Python support for returning tables in UDFs

### DISCUSSION THREAD

**Discussion by dthompson-maystreet** [duckdb/duckdb#9794] [General]
Hey team,

Can you provide a timeline for when Python UDFs can start returning table values? I understand currently scalar returns are supported but am wondering what kind of timeline you have for supporting table values being returned - so whether or not I need to learn C++!

Thank you folks,

Dave

**Comment by Tishj** [comment]
I don't think we have a timeline for it yet
You could try using `map` in the mean time 👍 
<https://duckdb.org/docs/archive/0.9.2/api/python/reference/>
`map(self: duckdb.duckdb.DuckDBPyRelation, ...)`
